libyul/Object.h

```diff
@@ -68,7 +68,15 @@ struct Object: ObjectNode
 
 	/// @returns the set of names of data objects accessible from within the code of
 	/// this object.
-	std::set<YulString> dataNames() const;
+	std::set<YulString> qualifiedDataNames() const;
+
+	/// @returns tuple consisting of
+	/// 	bool true if _qualifiedName is referring to data
+	/// 	vector of subIDs if possible to reach subobject with @_qualifiedName, empty vector otherwise
+	/// for "B.C" should return vector of two values if success (subId of B and subId of C in B)
+	/// in object "A" if called for "A.B" will return only one value (subId for B)
+	/// will return empty vector for @_qualifiedName that equals to object name
+	std::tuple<bool, std::vector<size_t>> pathToSubObject(std::string _qualifiedName) const;
```

Comment:
In every place where this function is called we're using `YulString` and then we need to call `.str()` on it so why not use it as the parameter type here?
```suggestion
	std::tuple<bool, std::vector<size_t>> pathToSubObject(YulString _qualifiedName) const;
```

Reply:
I think this is still relevant. Unless you think it's not a good idea but then just say so :)